torch/utils/data/distributed.py

```diff
@@ -36,23 +41,30 @@ def __init__(self, dataset, num_replicas=None, rank=None, shuffle=True):
         self.num_replicas = num_replicas
         self.rank = rank
         self.epoch = 0
-        self.num_samples = int(math.ceil(len(self.dataset) * 1.0 / self.num_replicas))
+
+        self._dataset_length = length
+        if not isinstance(length, int) or length <= 0:
```

Comment:
I think it's better to throw an error here. We want to explicitly check that, if length is not None, it is indeed an integer and indeed greater than 0, otherwise throw an error.